src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConstructorInitTests.cs

```diff
@@ -1258,5 +1258,86 @@ .maxstack  1
+                //     public C() : Garbage()
+                Diagnostic(ErrorCode.ERR_ConcreteMissingBody, "C").WithArguments("C.C()").WithLocation(4, 12),
+                // (4,18): error CS1018: Keyword 'this' or 'base' expected
+                //     public C() : Garbage()
+                Diagnostic(ErrorCode.ERR_ThisOrBaseExpected, "Garbage").WithLocation(4, 18),
+                // (4,18): error CS1002: ; expected
+                //     public C() : Garbage()
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "Garbage").WithLocation(4, 18),
+                // (4,18): error CS1520: Method must have a return type
+                //     public C() : Garbage()
+                Diagnostic(ErrorCode.ERR_MemberNeedsType, "Garbage").WithLocation(4, 18),
+                // (4,18): error CS0121: The call is ambiguous between the following methods or properties: 'C.C()' and 'C.C()'
+                //     public C() : Garbage()
+                Diagnostic(ErrorCode.ERR_AmbigCall, "").WithArguments("C.C()", "C.C()").WithLocation(4, 18)
+                );
+        }
+
+        [Fact, WorkItem(55797, "https://github.com/dotnet/roslyn/issues/55797")]
+        public void TwoParameterlessConstructors_2()
+        {
+            string source = @"
+public class C
+{
+    public C() : this()
+    {
+    }
+    public C()
+    {
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (4,18): error CS0121: The call is ambiguous between the following methods or properties: 'C.C()' and 'C.C()'
+                //     public C() : this()
+                Diagnostic(ErrorCode.ERR_AmbigCall, "this").WithArguments("C.C()", "C.C()").WithLocation(4, 18),
+                // (7,12): error CS0111: Type 'C' already defines a member called 'C' with the same parameter types
+                //     public C()
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "C").WithArguments("C", "C").WithLocation(7, 12)
+                );
+        }
+
+        [Fact, WorkItem(55797, "https://github.com/dotnet/roslyn/issues/55797")]
+        public void TwoParameterlessConstructors_3()
+        {
+            string source = @"
+public class C
+{
+    public C() : this()
+    {
+    }
+    public C2()
+    {
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (4,18): error CS0121: The call is ambiguous between the following methods or properties: 'C.C()' and 'C.C()'
+                //     public C() : this()
+                Diagnostic(ErrorCode.ERR_AmbigCall, "this").WithArguments("C.C()", "C.C()").WithLocation(4, 18),
+                // (7,12): error CS1520: Method must have a return type
+                //     public C2()
+                Diagnostic(ErrorCode.ERR_MemberNeedsType, "C2").WithLocation(7, 12)
+                );
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider testing with `struct` since the method that was failing is checking for value type constructors.